tests/RequestHeaderParserTest.php

```diff
@@ -113,6 +113,27 @@ public function testHeaderOverflowShouldEmitError()
         $this->assertSame('Maximum header size of 4096 exceeded.', $error->getMessage());
     }
 
+    public function testGuzzleRequestParseException()
+    {
+        $error = null;
+
+        $parser = new RequestHeaderParser();
+        $parser->on('headers', $this->expectCallableNever());
+        $parser->on('error', function ($message) use (&$error) {
+            $error = $message;
+        });
+
+        $this->assertSame(1, count($parser->listeners('headers')));
+        $this->assertSame(1, count($parser->listeners('error')));
+
+        $parser->feed("\r\n\r\n");
+
+        $this->assertInstanceOf('InvalidArgumentException', $error);
+        $this->assertSame('Invalid message', $error->getMessage());
+        $this->assertSame(0, count($parser->listeners('headers')));
+        $this->assertSame(0, count($parser->listeners('error')));
```

Comment:
Still agreeing with your previous comment but I rather make a new PR for that one as it feels related but out of scope for this PR